pkg/k8s/watchers/cilium_endpoint_batch.go

```diff
@@ -0,0 +1,202 @@
+func (k *K8sWatcher) endpointBatchAdded(ceb *cilium_v2a1.CiliumEndpointBatch) {
+
+	nodeIP := node.GetIPv4().String()
+	if option.Config.EnableIPv6 {
+		nodeIP = node.GetIPv6().String()
+	}
+
+	for i, ep := range ceb.Endpoints {
+		log.WithFields(logrus.Fields{
+			"CebName":          ceb.GetName(),
+			"CoreEndpointName": ep.Name,
+		}).Debug("New CEB added, Calling CoreEndpointUpdate")
+		c := k8s.ConvertCoreCiliumEndpointToTypesCiliumEndpoint(&ceb.Endpoints[i])
+		// Skip processing endpointupdate for local node CEPs
+		if c.Networking != nil && c.Networking.NodeIP == nodeIP {
+			continue
+		}
+		k.endpointUpdated(nil, c)
+	}
+}
+
+// endpointBatchUpdated invoked for modified CEBs, it compares old CEB and new CEB objects
+// determines below things
+// 1) any new core CEPs are packed in CEB
+// 2) any core CEPs are removed from CEB
+// 3) any existing core CEPs are modified in CEB
+// call endpointUpdated/endpointDeleted only for remote node CEPs.
+func (k *K8sWatcher) endpointBatchUpdated(oldCeb, newCeb *cilium_v2a1.CiliumEndpointBatch) {
+
+	nodeIP := node.GetIPv4().String()
+	if option.Config.EnableIPv6 {
+		nodeIP = node.GetIPv6().String()
+	}
+
+	oldCeps := make(map[string]*cilium_v2a1.CoreCiliumEndpoint)
+	for i, ep := range oldCeb.Endpoints {
+		oldCeps[ep.Name] = &oldCeb.Endpoints[i]
+	}
+
+	newCeps := make(map[string]*cilium_v2a1.CoreCiliumEndpoint)
+	for i, ep := range newCeb.Endpoints {
+		newCeps[ep.Name] = &newCeb.Endpoints[i]
+	}
+
+	// process removed entries from ceb. old ceb would have one or more stale cep
+	// entries. new ceb is source of truth.
+	for epName, oldEp := range oldCeps {
+		if _, exists := newCeps[epName]; !exists {
+			log.WithFields(logrus.Fields{
+				"CebName":          oldCeb.GetName(),
+				"CoreEndpointName": epName,
+			}).Debug("CEP deleted, Calling endpointDeleted")
+			c := k8s.ConvertCoreCiliumEndpointToTypesCiliumEndpoint(oldEp)
+			// Skip processing endpointupdate for local node CEPs
+			if c.Networking != nil && c.Networking.NodeIP == nodeIP {
+				continue
+			}
+			k.endpointDeleted(c)
+		}
+	}
+
+	// Process if any new items added into newCeb
+	for epName, newEp := range newCeps {
+		if _, exists := oldCeps[epName]; !exists {
+			log.WithFields(logrus.Fields{
+				"CebName":          oldCeb.GetName(),
+				"CoreEndpointName": epName,
+			}).Debug("CEB updated, Calling endpointUpdated")
+			c := k8s.ConvertCoreCiliumEndpointToTypesCiliumEndpoint(newEp)
+			// Skip processing endpointupdate for local node CEPs
+			if c.Networking != nil && c.Networking.NodeIP == nodeIP {
+				continue
+			}
+			k.endpointUpdated(nil, c)
+		}
+	}
+
+	// process if any cep entries changed from old to new
+	for epName, newCep := range newCeps {
+		if oldCep, exists := oldCeps[epName]; exists {
+			if oldCep.DeepEqual(newCep) {
+				continue
+			}
+			log.WithFields(logrus.Fields{
+				"CebName":          oldCeb.GetName(),
```

Comment:
this should probably be `newCeb.GetName()`?

Reply:
resolving this conversation, following up on previous comment.